asapdiscovery-data/asapdiscovery/data/util/utils.py

```diff
@@ -341,6 +341,126 @@ def cdd_to_schema(cdd_csv, out_json=None, out_csv=None):
     return compounds
 
 
+def cdd_to_schema_v2(
+    cdd_csv, target_prop, time_column=None, out_json=None, out_csv=None
+):
+    """
+    Convert a CDD-downloaded and filtered CSV file into a JSON file containing
+    a list[ExperimentalCompoundData]. CSV file should be the result of the
+    filter_molecules_dataframe function and must contain the following headers:
+    * name
+    * smiles
+    * achiral
+    * racemic
+    * <target property>
+
+    Parameters
+    ----------
+    cdd_csv : str
+        CSV file downloaded from CDD.
+    out_json : str, optional
+        JSON file to save to.
+    out_csv : str, optional
+        CSV file to save to.
+
+    Returns
+    -------
+    list[ExperimentalCompoundData]
+        The parsed list of ExperimentalCompoundData objects.
+    """
+
+    # Load and remove any straggling compounds w/o SMILES data
+    df = pandas.read_csv(cdd_csv)
+
+    # Check that all required columns are present
+    reqd_cols = [
+        "name",
+        "smiles",
+        "achiral",
+        "racemic",
+        "semiquant",
+        target_prop,
+    ]
+    missing_cols = [c for c in reqd_cols if c not in df.columns]
+    if len(missing_cols) > 0:
+        raise ValueError(f"Required columns not present in CSV file: {missing_cols}. ")
+
+    # Make extra sure nothing snuck by
+    idx = df["smiles"].isna()
+    logging.debug(f"Removing {idx.sum()} entries with no SMILES", flush=True)
+    df = df.loc[~idx, :]
+
+    # Fill semi-qunatitative data with the mean of others WHAT TO DO FOR regular scalar?
+    df.loc[df["semiquant"], target_prop] = df.loc[~df["semiquant"], target_prop].mean()
```

Comment:
in the original function, this was just for the experimental error. don't need to fill anything for the actual values, since we should also be checking if they're semi-quantitative when we use them later

Reply:
So this line can be safely removed?